packages/grafana-ui/src/components/Table/Table.test.tsx

```diff
@@ -673,4 +673,34 @@ describe('Table', () => {
       expect(selected).toBeVisible();
     });
   });
+
+  describe('when mounted with table cell callback', () => {
+    it('the row/column index attributes should be in the document', async () => {
+      const callback = jest.fn();
+      getTestContext({
+        onCellClick: callback,
+      });
+      expect(getTable()).toBeInTheDocument();
+
+      const rows = within(getTable()).getAllByRole('row');
+      expect(rows).toHaveLength(5);
+
+      rows.forEach((row, index) => {
+        // First row is column headers
+        if (index !== 0) {
+          expect(row).toHaveAttribute('data-row-index');
+        }
+      });
+
+      const cells = within(getTable()).getAllByRole('cell');
+      cells.forEach((cell) => {
+        expect(cell).toHaveAttribute('data-column-index');
+      });
+
+      // e.target.role is undefined so the function doesn't actually fire in tests :/
```

Comment:
Maybe try `fireEvent` with a custom Event?

Reply:
Same problem, the mocked event doesn't seem to have the expected props
<img width="1307" alt="image" src="https://github.com/grafana/grafana/assets/109082771/54d2c027-195b-45b0-b38c-b7e7181942d6">

Reply:
You should be able to [pass](https://testing-library.com/docs/dom-testing-library/api-events/#fireevent) an `Event` (or mocked Event) with the expected attributes like `fireEvent(node, yourEvent)`.

Reply:
Yeah I can't to figure it out, none of the mouseEvent options seem to allow me to set the role of the target.